torch/csrc/Module.cpp

```diff
@@ -1704,6 +1705,30 @@ Call this whenever a new thread is created in order to propagate values from
         return torch::should_allow_numbers_as_tensors(name);
       });
 
+  // FIXME(crcrpar): Better have `at::ScalarType` get mapped to `torch.dtype`
+  // Currently I see the second item of the key is displayed as
+  // e.g. `torch._C._te.ScalarType at 0x7fcf318adab0`
+  using _DeviceDtypeKey = std::pair<at::Device, std::string>;
+  using _FlatMap = std::unordered_map<
+      _DeviceDtypeKey,
+      at::native::TensorsAndIndicesT,
+      at::native::ParamsHash<_DeviceDtypeKey>>;
+  py_module.def(
+      "_group_tensors_by_device_and_dtype",
+      [](const std::vector<std::vector<c10::optional<at::Tensor>>>&
+             nested_tensorlist,
+         const bool with_indices) {
+        _FlatMap map;
+        for (const auto& iter :
+             at::native::group_tensors_by_first_tensors_device_and_dtype(
+                 nested_tensorlist, with_indices)) {
+          const auto scalar_type =
+              torch::utils::getDtypeNames(iter.first.second).first;
```

Comment:
This line is a bit crazy haha--to make sure I understand, torch::utils::getDtypeNames takes in a dtype and returns its names in terms of strings?

Maybe it'll be useful to define the type of iter in line 1722 instead of using auto& on it for clarity.